src/dashboard/src/installer/steps.py

```diff
@@ -106,6 +109,17 @@ def download_fpr_rules():
 
 
 def setup_pipeline_in_ss(use_default_config=False):
+    # Check if pipeline is already registered on SS
+    dashboard_uuid = helpers.get_setting('dashboard_uuid')
+    try:
+        storage_service._get_pipeline(dashboard_uuid)
+    except Exception:
+        logger.warning("SS inaccessible or pipeline not registered.")
+    else:
+        # If pipeline is already registered on SS, then exit
+        logger.warning("This pipeline is already configured on SS.")
+        return
```

Comment:
`_get_pipeline` only raises an exception when something wrong happened but when the pipeline is missing you should get a `None` value instead. Because of this I think L112-L121 could probably be better written as follows:

```python
    dashboard_uuid = helpers.get_setting('dashboard_uuid')
    pipeline = storage_service._get_pipeline(dashboard_uuid)
    if pipeline is not None:
        logger.warning("This pipeline is already configured on SS.")
        return
```

The exception isn't caught on purpose. If that was the case I think it's better to let the exception be raised.

Reply:
Also if you want to do this right I think that `_get_pipeline` should be renamed as `get_pipeline`. The underscore is a convention used to name function when they're private but if you start using it here then it's not private anymore.
When renamed, make sure that other uses of the function are updated.

Reply:
Hi Jesus,

Thanks very much for your suggestions. I have tested your second suggestion and it fails when upgrading after a MCP and SS database drop:

```
Retrying entries that fail because of foreign keys
maxLastUpdate at end 2014-01-20T15:44:12
WARNING   2018-01-09 12:30:01  archivematica.common:storageService:_get_pipeline:122:  This Archivematica instance is not registered with the storage service or has been disabled.
WARNING   2018-01-09 12:30:01  archivematica.common:storageService:_get_pipeline:125:  Error fetching pipeline
Traceback (most recent call last):
  File "/src/archivematicaCommon/lib/storageService.py", line 123, in _get_pipeline
    response.raise_for_status()
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 851, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 404 Client Error: NOT FOUND
Traceback (most recent call last):
  File "/src/dashboard/src/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/src/dashboard/src/installer/management/commands/install.py", line 58, in handle
    setup_pipeline_in_ss(use_default_config=True)
  File "/src/dashboard/src/installer/steps.py", line 115, in setup_pipeline_in_ss
    pipeline = storage_service._get_pipeline(dashboard_uuid)
  File "/src/archivematicaCommon/lib/storageService.py", line 123, in _get_pipeline
    response.raise_for_status()
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 851, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: NOT FOUND
Makefile:51: recipe for target 'bootstrap-dashboard' failed
```

I have updated this branch with the other changes you have suggested, so it is ready for CR again.

Reply:
You're right, `raise_for_status()` is used causing the call to raise a `HTTPError` after that 404.

Thank you!